src/main/webapp/user_profile_script.js

```diff
@@ -12,17 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function deleteRow(row) {
-    if (confirm("Are you sure that you want to delete this already published task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("need-help").deleteRow(rowIndex);
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info);
+    return info;
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
+        window.alert("You can only delete an 'OPEN' task.")
+    } else {
+        if (confirm("Are you sure that you want to delete the task?")) {
+            const queryURL = "/tasks?key=" + keyString;
+            const request = new Request(queryURL, {method: "DELETE"});
+            const response = await fetch(request);
```

Comment:
Should we hide the row after the delete succeeds? Right now the task stays on the page until I refresh.

Reply:
In the new commit, I force the page to load the need help table again after the deletion occurs. Now the task should be removed once the user deleted the task.